tensordict/prototype/tensorclass.py

```diff
@@ -123,14 +148,34 @@ def __torch_function__(
         args: tuple[Any, ...] = (),
         kwargs: dict[str, Any] | None = None,
     ) -> Callable:
+        if (
+            func not in tc_handled_functions and func not in _TD_PASS_THROUGH
+        ) or not all(issubclass(t, (Tensor, cls)) for t in types):
+            return NotImplemented
         if kwargs is None:
             kwargs = {}
-        if func not in td_handled_functions or not all(
-            issubclass(t, (Tensor, cls)) for t in types
-        ):
-            return NotImplemented
+        if func in _TD_PASS_THROUGH:
+            if len(args) > 0:
+                input_ = args[0]
+            else:
+                input_ = kwargs["input"] if "input" in kwargs else kwargs["tensors"]
```

Comment:
```suggestion
                input_ = kwargs.get("input", kwargs["tensors"])
```

Reply:
Oh I actually had the same idea! This is changed in the latest commit